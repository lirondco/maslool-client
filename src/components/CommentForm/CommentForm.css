.CommentForm {
  margin-bottom: 2%;
  padding: 1%;
  border: #524c57 solid;
  border-radius: 10px;
  box-shadow: 5px 5px 2.5px grey;
  margin: 0 !important;
  background-color: lightblue;
}
