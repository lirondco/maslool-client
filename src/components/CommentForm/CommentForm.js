import React, { Component } from "react";
import TrailContext from "../../contexts/TrailContext";
import TrailsApiService from "../../services/trails-api-service";
import { Textarea, Button } from "../Utils/Utils";
import "./CommentForm.css";

export default class CommentForm extends Component {
  static contextType = TrailContext;

  handleSubmit = (ev) => {
    ev.preventDefault();
    const { trail } = this.context;
    const { content } = ev.target;
    TrailsApiService.postComment(trail.id, content.value)
      .then(this.context.addComment)
      .then(() => {
        content.value = "";
      })
      .catch(this.context.setError);
  };

  render() {
    return (
      <form className="CommentForm" onSubmit={this.handleSubmit}>
        <div className="content">
          <Textarea
            required
            aria-label="Type a comment..."
            name="content"
            id="content"
            placeholder="Type a comment..."
          />
        </div>
        <Button type="submit">Post Comment</Button>
      </form>
    );
  }
}
