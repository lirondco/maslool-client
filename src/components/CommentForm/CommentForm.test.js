import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { withContext } from 'shallow-with-context'
import CommentForm from './CommentForm'

describe(`CommentForm component`, () => {
    it('renders the complete form', () => {
        const wrapper = shallow(<CommentForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('validates form on submit', () => {
        const context = { trail: { id: 999 } };
        const ComponentWithContext = withContext(CommentForm, context);
        const handleSubmit = jest.fn();
        const wrapper = shallow(
            <ComponentWithContext handleSubmit={handleSubmit}/>,
            { context }
        );
        const mockedEvent = { preventDefault: () => {}, target: { content: { value: 'Test' } } }
        wrapper.find('form').simulate('submit', mockedEvent)
      })
})
