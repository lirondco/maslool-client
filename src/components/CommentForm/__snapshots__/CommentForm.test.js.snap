// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm component renders the complete form 1`] = `
<form
  className="CommentForm"
  onSubmit={[Function]}
>
  <div
    className="content"
  >
    <Textarea
      aria-label="Type a comment..."
      id="content"
      name="content"
      placeholder="Type a comment..."
      required={true}
    />
  </div>
  <ForwardRef
    type="submit"
  >
    Post Comment
  </ForwardRef>
</form>
`;
